back/engines/commercial/nlp/app/services/nlp/api.rb

```diff
@@ -127,6 +127,21 @@ def cancel_task(task_id)
       resp.code
     end
 
+    # @param [Array<String>] task_ids
+    # @return [Integer] HTTP status code
+    def cancel_tasks(task_ids)
+      body = { ids: task_ids }
+
+      resp = post(
+        "/v2/async_api/cancel",
+        body: body.to_json,
+        headers: authorization_header.merge('Content-Type' => 'application/json'),
```

Comment:
It seems all the methods in this class use the same headers and timeout. Maybe it would look prettier and would be easier to test if there was a method like `post_req`

```ruby
def cancel_tasks(task_ids)
  post_req("/v2/async_api/cancel", { ids: task_ids }).code
end

private

def post_req(path, body)
  post(path, body.to_json, ...)
end
```

But it's definitely not a requirement, just an idea.

Reply:
Yep, I know. That's something I had in the back of my head for some time. So many issues with this class... The thing is that sometimes there are slight differences and the NLP API can be a bit touchy. I'll see what I can do.